### ISSUE TITLE
Failure while polling the secret

### ISSUE THREAD

**Issue by maiakhoa** [jx3-gitops-repositories/jx3-terraform-eks#20]
I got this error from external-secret when running on EKS 1.21

`"payload":{"response":{"statusCode":500,"body":{"errors":["claim \"iss\" is invalid"]}}},"msg":"failure while polling the secret jx/nexus"}`

I guess we can fix this with this article 

`https://particule.io/en/blog/vault-1.21/`

**Comment by maiakhoa** [comment]
when I tried to run below to fix the issue, I got the error `Error writing data to auth/kubernetes/config: Put "https://127.0.0.1:8200/v1/auth/kubernetes/config": x509: certificate signed by unknown authority`. Anyone can help to advice

```
vault write auth/kubernetes/config \
  token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
  kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443" \
  kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
  issuer="$YOUR_PREVIOUSLY_COMPUTED_ISSUER \
```

**Comment by maiakhoa** [comment]
I can solve this by running below in vault instance

Getting vault token first

`export VAULT_TOKEN=$(kubectl get secrets vault-unseal-keys  -n jx-vault -o jsonpath={.data.vault-root} | base64 --decode)`

Then run below in vault pod

```
export VAULT_SKIP_VERIFY=1
vault init
vault write auth/kubernetes/config token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443" kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt issuer="{{ISSUER}}"
```

Then wait for jx-git-operator working back, it can fix the issue

I hope we can upgrade bank-vault instance to the latest one. I believe they fixed the issue related to issuer on 1.21

**Comment by mrmarcsmith** [comment]
I just experienced the same issue on EKS 1.15.46. I would say this issue is not resolved. can we reopen it?
here is my setup
main.tf
```
module "eks-jx" {
  source               = "jenkins-x/eks-jx/aws"
  version              = "1.15.46"
...
}
```
variables.tf
```
variable "cluster_version" {
  description = "Kubernetes version to use for the EKS cluster."
  type        = string
  default     = "1.21"
}
variable "nginx_chart_version" {
  type        = string
  description = "nginx chart version"
  default     = "3.12.0"
}
```
<img width="1174" alt="Screen Shot 2021-09-16 at 9 23 15 PM" src="https://user-images.githubusercontent.com/5037688/133723793-aa683199-68e8-4ee3-801d-27a96b0cf871.png">

**Comment by mrmarcsmith** [comment]
<img width="1560" alt="Screen Shot 2021-09-16 at 9 28 38 PM" src="https://user-images.githubusercontent.com/5037688/133724590-5ed2c4b6-44a4-4add-8621-a23b0cd6be4e.png">

**Comment by maiakhoa** [comment]
@mrmarcsmith after you update to Vault pod, you have to restart jx-git-operator, but it's a trick.

I changed the helm file from infrastructure git to my own helm chart to upgrade vault-instance, I see latest version of vault-instance has fixed this. You can use my own clone helm and just upgrade vault-instance to 1.41.1

```
resource "helm_release" "vault-instance" {
  count      = local.create_vault_resources ? 1 : 0
  name       = "vault-instance"
  chart      = "vault-instance"
  namespace  = "jx-vault"
  repository = "https://maiakhoa.github.io/vault-instance"
  version    = "1.4.0"
  depends_on = [helm_release.vault-operator]
}
```

Please notice that this fix just for EKS 1.21

**Comment by mrmarcsmith** [comment]
@maiakhoa Thank you for the comment. While I appreciate the workaround, I would really like to stay within a "standard jx3 setup" from an auditing, security, and sustainability standpoint. @ankitm123 how long until the upgrade @maiakhoa has pointed out is applied to the main infra repo? I have tested up to 1.15.47 and it still fails. The root issue appears to be a change is 1.21
